backend/app/providers/telegram.go

```diff
@@ -0,0 +1,65 @@
+package providers
+
+// This provider provides common logic, reused by Telegram auth and notifications verification
```

Comment:
From what you explained me earlier, it seems what the real reason why this Dispatch thing even needed is the nature of tg API requiring a common/single event loop, and we need it for both auth and notify. This is not that clear from the comment about "common logic".